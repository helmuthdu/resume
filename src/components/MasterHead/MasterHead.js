// @flow
import React from 'react';
import { Button, Container, Divider, Header, Segment } from 'semantic-ui-react';
import { menuItems } from '../../layouts/DefaultLayout';
import HeaderMenu from '../HeaderMenu/HeaderMenu';

export const MasterHead = ({ pathname }: any) => (
  <Segment vertical inverted textAlign="center" className="masthead">
    <HeaderMenu pathname={pathname} items={menuItems} pointing secondary inverted visible />
    <Container text>
      <Header inverted as="h1">
        Helmuth Saatkamp
      </Header>
      <Divider hidden />
      <Header inverted as="h2">
        Web Developer
      </Header>
      <Button
        primary
        inverted
        size="large"
        as="a"
        href={`${process.env.PUBLIC_URL || ''}/docs/helmuth_saatkamp_resume_en.pdf`}>
        Download My CV
      </Button>
    </Container>
  </Segment>
);

export default MasterHead;
