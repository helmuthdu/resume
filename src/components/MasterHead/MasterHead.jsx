import PropTypes from 'prop-types';
import React from 'react';
import { Button, Container, Divider, Header, Segment } from 'semantic-ui-react';
import { menuItems } from '../../layouts/DefaultLayout';
import HeaderMenu from '../HeaderMenu/HeaderMenu';

export const MasterHead = ({ pathname }) => (
  <Segment vertical inverted textAlign="center" className="masthead">
    <HeaderMenu pathname={pathname} items={menuItems} pointing secondary inverted visible />
    <Container text>
      <Header inverted as="h1">
        Helmuth Saatkamp
      </Header>
      <Divider hidden />
      <Header inverted as="h2">
        Software Engineer
      </Header>
      <Button
        color={'blue'}
        inverted
        size="large"
        as="a"
        href="https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf">
        Download My CV
      </Button>
    </Container>
  </Segment>
);

MasterHead.propTypes = {
  pathname: PropTypes.string.isRequired
};

export default MasterHead;
