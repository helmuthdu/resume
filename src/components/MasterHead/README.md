# MasterHead component

MasterHead Section.

## Source

    <MasterHead id={name}/>
