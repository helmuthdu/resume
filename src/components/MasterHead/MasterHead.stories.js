const withReadme = require('storybook-readme/with-readme').default;
const AboutReadme = require('./README.md');

import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import MasterHead from './MasterHead';

storiesOf('MasterHead', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(AboutReadme))
  .add('default', () => {
    return <MasterHead />;
  });
