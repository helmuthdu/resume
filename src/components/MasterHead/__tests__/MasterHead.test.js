import { shallow } from 'enzyme';
import React from 'react';
import { MasterHead } from '../MasterHead';

describe('component -> MasterHead', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<MasterHead pathname="/" />);
    expect(wrapper).toMatchSnapshot();
  });
});
