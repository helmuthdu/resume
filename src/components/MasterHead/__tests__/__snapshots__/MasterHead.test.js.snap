// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component -> MasterHead should render correctly 1`] = `
<Segment
  className="masthead"
  inverted={true}
  textAlign="center"
  vertical={true}
>
  <Connect(HeaderMenu)
    inverted={true}
    items={
      Array [
        Object {
          "exact": true,
          "icon": "home",
          "inverted": true,
          "name": "Home",
          "path": "/",
        },
        Object {
          "exact": false,
          "icon": "info circle",
          "name": "About",
          "path": "#about",
        },
        Object {
          "exact": false,
          "icon": "newspaper",
          "name": "Resume",
          "path": "#resume",
        },
        Object {
          "exact": false,
          "icon": "newspaper",
          "name": "Portfolio",
          "path": "#portfolio",
        },
        Object {
          "exact": false,
          "icon": "newspaper",
          "name": "Contact",
          "path": "#contact",
        },
      ]
    }
    pathname="/"
    pointing={true}
    secondary={true}
    visible={true}
  />
  <Container
    text={true}
  >
    <Header
      as="h1"
      inverted={true}
    >
      Helmuth Saatkamp
    </Header>
    <Divider
      hidden={true}
    />
    <Header
      as="h2"
      inverted={true}
    >
      Web Developer
    </Header>
    <Button
      as="a"
      href="https://github.com/helmuthdu/resume/raw/master/public/docs/helmuth_saatkamp_resume_en.pdf"
      inverted={true}
      primary={true}
      size="large"
    >
      Download My CV
    </Button>
  </Container>
</Segment>
`;
